sql/2019/media/04_09b.sql

```diff
@@ -16,8 +16,10 @@ FROM
     WHERE
       date = '2019-07-01' AND
       firstHtml AND
-      (REGEXP_CONTAINS(body, r'(?is)<meta[^><]*Accept-CH\b') OR
-        REGEXP_CONTAINS(respOtherHeaders, r'(?is)Accept-CH = '))
+      (
+        REGEXP_CONTAINS(body, r'(?is)<meta[^><]*Accept-CH\b') OR
+        REGEXP_CONTAINS(respOtherHeaders, r'(?is)Accept-CH = ')
+      )
```

Comment:
This should use K&R

Reply:
So you mean move the opening bracket up to previous line?

```sql
WHERE
      date = '2019-07-01' AND
      firstHtml AND (
        REGEXP_CONTAINS(body, r'(?is)<meta[^><]*Accept-CH\b') OR
        REGEXP_CONTAINS(respOtherHeaders, r'(?is)Accept-CH = ')
      )
```

Personally I find that harder to read. We have three distinct `AND` clauses but decide to put the start of the third one at the end of line 2.

To me this is clearer:

```sql
WHERE
      date = '2019-07-01' AND
      firstHtml AND
      (
        REGEXP_CONTAINS(body, r'(?is)<meta[^><]*Accept-CH\b') OR
        REGEXP_CONTAINS(respOtherHeaders, r'(?is)Accept-CH = ')
      )
```

But the linter is fine with either (through prefers the latter, hence why the auto fix moved to that when it had to fix the file for other reasons) so can correct to that if that's the preference.

Reply:
Oh and the wikipedia entry has this (emphasis my own):

> A multi-statement block inside a function, however, has its opening brace on the same line **as its control clause** while the closing brace remains on its own line

So while it makes sense to have this:

```js
while {
  ...
}
```

as the `{` is controlled by the `while` and is part of that clause, I'm less convinced about independent clauses.

Surely you wouldn't do this as arguments to a function that use brackets, when they are spread over separate lines:

```js
myfunction(
  a, (
    b+c
  ),
  d
) {
```

But instead would do this?:

```js
myfunction(
  a,
  (b+c),
  d
) {
```

And that's exactly what we have here: independent clauses separated by an `AND` (instead of a comma).

Reply:
So you're saying if they couldn't fit on one line you'd do this:

```js
myfunction(
  a, (
    b +
    c
  ),
  d
) {
```

That's just weird to me, and not K&R, but OK can do that if we want.

Reply:
Or should it be like this?

```js
myfunction(
  a,
  (b +
    c),
  d
) {
```

Confused as to what it wanted here?